tubular/mapping.py

```diff
@@ -163,13 +166,25 @@ def __init__(self, mappings, **kwargs):
         BaseMappingTransformer.__init__(self, mappings=mappings, **kwargs)
 
     def transform(self, X):
-        """Transfrom the input data X according to the mappings in the mappings attribute dict.
+        """Transform the input data X according to the mappings in the mappings attribute dict.
 
         This method calls the BaseMappingTransformMixin.transform. Note, this transform method is
         different to some of the transform methods in the nominal module, even though they also
         use the BaseMappingTransformMixin.transform method. Here, if a value does not exist in
         the mapping it is unchanged.
 
+        Due to the way pd.Series.map works, mappings can result in column dtypes changing,
+        sometimes unexpectedly. If the result of the mappings is a dtype that doesn't match
+        the original dtype, or the dtype of the values provided in the mapping a warning
+        will be raised. This normally results from an incomplete mapping being provided,
+        or a mix of dtypes causing pandas to default to the object dtype.
+
+        For columns with a 'category' dtype, conventional type comparisons do not work as they
+        category dtypes reflect the possible values on the category. This means that although
+        a mapping can maintain a category dtype, if the values in the column change it will
+        be registered as a different dtype. To address this a check is performed before the
```

Comment:
I think just: "The warning will not be raised in this case" is clearer - this description led me to look for a pre-transform check for category dtype

Reply:
agreed and changed to suggested simpler explanation in [4632f52](https://github.com/lvgig/tubular/pull/54/commits/4632f527a938d1962110eda5af823b52e7434caa)